src/Controls/src/Core/AppThemeBinding.cs

```diff
@@ -41,34 +42,101 @@ internal override void Unapply(bool fromBindingContextChanged = false)
 			DetachEvents();
 
 			base.Unapply(fromBindingContextChanged);
-			_weakTarget = null;
-			_targetProperty = null;
+			TargetProperty = null;
+			Target = null;
 		}
 
 		void OnRequestedThemeChanged(object sender, AppThemeChangedEventArgs e)
 			=> ApplyCore(true);
```

Comment:
Does this generate on background threads? I'm wondering if we should have a version of this that always fires on the UIThread.  Otherwise, every single `AppThemeBinding` is going to dispatch changes individually.  If we just ensure that on `Window` there's a version of this (or maybe this one already does) that fires on the UIThread then all the `AppThemeBindings` changes can just happen at once.